prog/app.go

```diff
@@ -18,14 +18,21 @@ import (
 )
 
 // Router creates the mux for all the various app components.
-func router(c app.Collector) *mux.Router {
+func router(c app.Collector) http.Handler {
 	router := mux.NewRouter()
-	app.RegisterTopologyRoutes(c, router)
 	app.RegisterReportPostHandler(c, router)
 	app.RegisterControlRoutes(router)
 	app.RegisterPipeRoutes(router)
+	app.RegisterTopologyRoutes(c, router)
 	router.Methods("GET").PathPrefix("/").Handler(http.FileServer(FS(false)))
-	return router
+
+	// need to handle some routes in a completely custom way, because
+	// https://github.com/weaveworks/scope/issues/804
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !app.HandleNode(c, w, r) {
+			router.ServeHTTP(w, r)
+		}
+	})
```

Comment:
Don't like that this is in 2 places. Maybe extract it somewhere into a function with the explanation there. Could just inline it into `app.HandleNode`, maybe.